### PR TITLE
Dev sir 1229 document contract

### PR DESCRIPTION
**Pull request by aushakov**

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -33,7 +32,7 @@ jobs:
       - name: Publish to Registry
         uses: swisschain/Publish-Docker-Github-Action@master
         with:
-          name: swisschains/sirius-vault-hsm
+          name: swisschains/sirius-vault-hsm:latest
```
Comment (KonstantinRyazantsev) [inline .github/workflows/gradle.yml:35]: why not `dev`?

File: src/main/java/io/swisschain/contracts/RequestContext.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  public Instant getTimestamp() {
+    return timestamp;
+  }
+
+  public void setTimestamp(Instant timestamp) {
+    this.timestamp = timestamp;
+  }
+
+  public boolean equal(RequestContext requestContext) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/RequestContext.java:52]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

File: src/main/java/io/swisschain/contracts/SourceAddress.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  public String getGroup() {
+    return group;
+  }
+
+  public void setGroup(String group) {
+    this.group = group;
+  }
+
+  public boolean equal(SourceAddress sourceAddress) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/SourceAddress.java:40]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

File: src/main/java/io/swisschain/contracts/TransferContext.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  public RequestContext getRequestContext() {
+    return requestContext;
+  }
+
+  /** Sets the client request details. */
+  public void setRequestContext(RequestContext requestContext) {
+    this.requestContext = requestContext;
+  }
+
+  public boolean equal(TransferContext transferContext) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/TransferContext.java:128]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

File: src/main/java/io/swisschain/contracts/TransferDetails.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  public TransferContext getTransferContext() {
+    return transferContext;
+  }
+
+  /** Sets the transfer detailed information and initial client request. */
+  public void setTransferContext(TransferContext transferContext) {
+    this.transferContext = transferContext;
+  }
+
+  public boolean equal(TransferDetails transferDetails) {
```
Comment (dmitrykrasovskih) [inline src/main/java/io/swisschain/contracts/TransferDetails.java:117]: Please use autogenerated equals and hashCode - they should be synchronized with each other.

**Review by dmitrykrasovskih — CHANGES_REQUESTED** [review]

**Pull request merged**